src/coreclr/nativeaot/Runtime/DebugHeader.cpp

```diff
@@ -0,0 +1,279 @@
+    uint32_t FieldOffset;
+    uint32_t ResevedPadding;
+};
+
+struct GlobalValueEntry
+{
+    const char *Name;
+    const void *Address;
+};
+
+// This size should be one bigger than the number of entries since a null entry
+// signifies the end of the array.
+static constexpr size_t DebugTypeEntriesArraySize = 96;
+static DebugTypeEntry s_DebugEntries[DebugTypeEntriesArraySize];
+
+// This size should be one bigger than the number of entries since a null entry
+// signifies the end of the array.
+static constexpr size_t GlobalEntriesArraySize = 6;
+static GlobalValueEntry s_GlobalEntries[GlobalEntriesArraySize];
+
+// This structure is part of a in-memory serialization format that is used by diagnostic tools to
+// reason about the runtime. As a contract with our diagnostic tools it must be kept up-to-date
+// by changing the MajorVersion when breaking changes occur. If you are changing the runtime then
+// you are responsible for understanding what changes are breaking changes. You can do this by
+// reading the specification (Documentation\design-docs\diagnostics\ProcessMemoryFormatSpec.md)
+// to understand what promises the runtime makes to diagnostic tools. Any change that would make that
+// document become inaccurate is a breaking change.
+//
+// If you do want to make a breaking change please coordinate with diagnostics team as breaking changes
+// require debugger side components to be updated, and then the new versions will need to be distributed
+// to customers. Ideally you will check in updates to the runtime components, the debugger parser
+// components, and the format specification at the same time.
+//
+// Although not guaranteed to be exhaustive, at a glance these are some potential breaking changes:
+//   - Removing a field from this structure
+//   - Reordering fields in the structure
+//   - Changing the data type of a field in this structure
+//   - Changing the data type of a field in another structure that is being refered to here with
+//       the offsetof() operator
+//   - Changing the data type of a global whose address is recorded in this structure
+//   - Changing the meaning of a field or global refered to in this structure so that it can no longer
+//     be used in the manner the format specification describes.
+struct DotNetRuntimeDebugHeader
+{
+    // The cookie serves as a sanity check against process corruption or being requested
+    // to treat some other non-.Net module as though it did contain the coreRT runtime.
+    // It can also be changed if we want to make a breaking change so drastic that
+    // earlier debuggers should treat the module as if it had no .Net runtime at all.
+    // If the cookie is valid a debugger is safe to assume the Major/Minor version fields
+    // will follow, but any contents beyond that depends on the version values.
+    // The cookie value is currently set to 0x4E 0x41 0x44 0x48 (NADH in ascii)
+    const uint8_t Cookie[4] = { 0x4E, 0x41, 0x44, 0x48 };
```

Comment:
Nit: Does NADH stand for 'Native AOT Debug Header'? If we changed the name elsewhere maybe this should change too for naming consistency.

Reply:
That is definitely what it stands for 😄. I can change it to DNDH for DotNetDebugHeader.